src/cascadia/TerminalSettingsModel/ApplicationState.cpp

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "ApplicationState.h"
+#include "CascadiaSettings.h"
+#include "ApplicationState.g.cpp"
+
+#include "JsonUtils.h"
+#include "FileUtils.h"
+
+namespace Microsoft::Terminal::Settings::Model::JsonUtils
+{
+    // This trait exists in order to serialize the std::unordered_set for GeneratedProfiles.
+    template<typename T>
+    struct ConversionTrait<std::unordered_set<T>>
+    {
+        std::unordered_set<T> FromJson(const Json::Value& json) const
+        {
+            ConversionTrait<T> trait;
+            std::unordered_set<T> val;
+            val.reserve(json.size());
+
+            for (const auto& element : json)
+            {
+                val.emplace(trait.FromJson(element));
+            }
+
+            return val;
+        }
+
+        bool CanConvert(const Json::Value& json) const
+        {
+            ConversionTrait<T> trait;
+            return json.isArray() && std::all_of(json.begin(), json.end(), [trait](const auto& json) -> bool { return trait.CanConvert(json); });
+        }
+
+        Json::Value ToJson(const std::unordered_set<T>& val)
+        {
+            ConversionTrait<T> trait;
+            Json::Value json{ Json::arrayValue };
+
+            for (const auto& key : val)
+            {
+                json.append(trait.ToJson(key));
+            }
+
+            return json;
+        }
+
+        std::string TypeDescription() const
+        {
+            return fmt::format("std::unordered_set<{}>", ConversionTrait<GUID>{}.TypeDescription());

```

Comment:
This message is intended for display to the user, but at the same time, the state deserializer is meant to be silent and discard-on-failure :smile:

Reply:
What would you suggest? Should I remove the implementation of this function and return an empty string?

Reply:
my recommendation would be something less C++y and more javascripty. "array (of {})" or something similar. I don't wholly mind if you leave it like this

Reply:
or even like, "list of guids"